File: app/src/main/java/vn/loitp/app/activity/customviews/recyclerview/recyclertablayout/years/DemoYearsPagerAdapter.java
package vn.loitp.app.activity.customviews.recyclerview.recyclertablayout.years;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.viewpager.widget.PagerAdapter;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

import loitp.basemaster.R;

public class DemoYearsPagerAdapter extends PagerAdapter {

    private List<String> mItems = new ArrayList<>();

    public DemoYearsPagerAdapter() {
    }

    @NotNull
    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        View view = LayoutInflater.from(container.getContext())
                .inflate(R.layout.layout_page, container, false);

        TextView textView = view.findViewById(R.id.title);
        textView.setText("Page: " + mItems.get(position));
        container.addView(view);

        return view;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, @NotNull Object object) {
        container.removeView((View) object);
    }

    @Override
    public int getCount() {
        return mItems.size();
    }

    @Override
    public boolean isViewFromObject(@NotNull View view, @NotNull Object object) {
        return view == object;
    }

    @Override
    public String getPageTitle(int position) {
        return mItems.get(position);
    }

    public void addAll(List<String> items) {
        mItems = new ArrayList<>(items);
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/recyclerview/recyclertablayout/years/DemoYearsPagerAdapter.kt
package vn.loitp.app.activity.customviews.recyclerview.recyclertablayout.years

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.viewpager.widget.PagerAdapter
import loitp.basemaster.R
import java.util.*

class DemoYearsPagerAdapter : PagerAdapter() {

    private var mItems: List<String> = ArrayList()

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view = LayoutInflater.from(container.context)
                .inflate(R.layout.item_recycler_tablayout_page, container, false)

        val textView = view.findViewById<TextView>(R.id.title)
        textView.text = "Page: " + mItems[position]
        container.addView(view)

        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, ob: Any) {
        container.removeView(ob as View)
    }

    override fun getCount(): Int {
        return mItems.size
    }

    override fun isViewFromObject(view: View, ob: Any): Boolean {
        return view === ob
    }

    override fun getPageTitle(position: Int): String? {
        return mItems[position]
    }

    fun addAll(items: List<String>) {
        mItems = ArrayList(items)
    }
}
